### DISCUSSION TITLE
Machine user cannot push to protected `main` branch despite being granted push access

### DISCUSSION THREAD

**Discussion by DavidJacob** [community/community#39763] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I have been trying to figure this out for two days now, but I can't get it. I have a "machine user", let's call it `GitOpsBot`, which is responsible for handling the commits from my CI pipeline to my GitOps repo. I have generated a classic github token with all `repo` permissions to handle these workflows.

Some info about the `GitOpsBot`:
* it has a `Member` role in the organization
* it has an `Admin` role in the repository

I want to create a branch protection rule with the following criteria:
* all human users need to create a PR to merge any code to `main`
* **only** my `GitOpsBot` can push straight to `main`

With the configuration outlined below, the `GitOpsBot` **cannot** push to the `main` branch. If I disable the `Do not allow bypassing the above settings`, it **can** push to the `main` branch.

What is going on? Shouldn't the explicit listing of the user in the `Restrict who can push to matching branches` option allow the `GitOpsBot` to push straight to `main`?

Thanks for any help you can provide.

![2022-11-22_16-08](https://user-images.githubusercontent.com/4319427/203349350-21c56f50-bf87-4b56-a778-4d1839bf8eec.png)
![2022-11-22_16-00](https://user-images.githubusercontent.com/4319427/203347871-43c08adf-df5e-4d4d-a226-5384870e040a.png)
![2022-11-22_16-02](https://user-images.githubusercontent.com/4319427/203347887-650e902f-04e3-4beb-81da-e7a3f2bfc9e1.png)

**Comment by ossanna16** [comment]
Hi there @DavidJacob and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)
